### ISSUE TITLE
Loading and saving external data

### ISSUE THREAD

**Issue by shoyer** [mars-project/mars#105]
This might be a silly question, but how do you load and save data with mars?

With dask.array, the main APIs are `dask.array.from_array` and `dask.array.store`. These allow for a great deal of flexibility, users can pass in arbitrary Python objects that support the `__getitem__` and `__setitem__` protocols.

I am intrigued about the idea of testing with mars with [xarray](http://xarray.pydata.org), but these APIs would be pretty essential for us. In xarray, we use these to support IO to various file formats (e.g., HDF5, netCDF, zarr).

**Comment by chineking** [comment]
I want to bring more backgrounds that for the inner projects which use Mars, we write the interface like `mars.tensor.read_files(distributed_storage_directory)`, this requires users to write a MapReduce-like program to process the data into different directories in the distributed storage directory, each directory contains files with parquet or some columnar storage format, stored with `i, j, k, value` schema. 

We were stuck in how to provide a commonly used interface which need to be as compatible as possible.

**Comment by shoyer** [comment]
Indeed, file formats for multi-dimensional arrays are challenging. I think the main virtue of dask's approach here is that it's quite flexible, and allow reusing existing code for working with files. The downside is that now distributed workers need to know how to run arbitrary Python code from users (but this can also be quite useful, as a form of foreign function interface).
